src/overlays/actors/ovl_En_Encount2/z_en_encount2.c

```diff
@@ -22,19 +36,317 @@ const ActorInit En_Encount2_InitVars = {
+                    // rock spawn pos X,Z at a random position near player
+                    tempVec2X = Rand_CenteredFloat(70.0f) + player->actor.world.pos.x;
+                    tempVec2Z = Rand_CenteredFloat(70.0f) + player->actor.world.pos.z;
+                }
+            }
+            spawnedRock =
+                (EnFireRock*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_FIRE_ROCK,
+                                                tempVec2X, tempVec1Y, tempVec2Z, 0, 0, 0, spawnedRockType);
+            if (spawnedRock != NULL) {
+                spawnedRock->spawner = this;
+                this->numSpawnedRocks++;
+                return;
+            }
+            // "☆☆☆☆☆ Can't occur! ☆☆☆☆☆"
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n\n" VT_RST);
+        }
+    }
+}
+
+void EnEncount2_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnEncount2* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    if (this->deathMountainSpawnerTimer != 0) {
+        this->deathMountainSpawnerTimer--;
+    }
+    if (this->timerBetweenRockSpawns != 0) {
+        this->timerBetweenRockSpawns--;
+    }
+    if (this->particleSpawnTimer != 0) {
+        this->particleSpawnTimer--;
+    }
+    this->actionFunc(this, globalCtx);
+    EnEncount2_ParticleUpdate(this, globalCtx);
+    if (!this->isNotDeathMountain) {
+        this->unk17C = this->envEffectsTimer / 60.0f;
+        this->unk160 = this->unk17C * -50.0f;
+        globalCtx->envCtx.unk_8C[0][0] = (s16)this->unk160 * -1.5f;
+        globalCtx->envCtx.unk_8C[0][1] = globalCtx->envCtx.unk_8C[0][2] = this->unk160;
+        this->unk168 = this->unk17C * -20.0f;
+        globalCtx->envCtx.unk_8C[1][0] = (s16)this->unk168 * -1.5f;
+        globalCtx->envCtx.unk_8C[1][1] = globalCtx->envCtx.unk_8C[1][2] = this->unk168;
+        this->unk170 = this->unk17C * -50.0f;
+        globalCtx->envCtx.unk_9E = this->unk170;
+        globalCtx->envCtx.unk_8C[2][0] = (u8)((160.0f - globalCtx->envCtx.unk_CF[0]) * this->unk17C);
+        globalCtx->envCtx.unk_8C[2][1] = (u8)((160.0f - globalCtx->envCtx.unk_CF[1]) * this->unk17C);
+        globalCtx->envCtx.unk_8C[2][2] = (u8)((150.0f - globalCtx->envCtx.unk_CF[2]) * this->unk17C);
+    }
+}
+
+void EnEncount2_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnEncount2* this = THIS;
+    EnEncount2_ParticleDraw(&this->actor, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Encount2/func_80A07CA4.s")
+void EnEncount2_ParticleInit(EnEncount2* this, Vec3f* particlePos, f32 scale) {
+    EnEncount2Particle* particle = this->particles;
+    s16 i;
```

Comment:
in general we stay away from the word particle and use effect, cause alot of these things that have the same structure dont really classify as particles. depends on the context though I guess
what is this actually drawing? its not a rock is it?

Reply:
It's the small particles of burning rocks that fall out of the sky in collapse and DMT, I'm not sure what a good name would be